tests/study_tests/test_study.py

```diff
@@ -1010,3 +1010,71 @@ def test_tell_pruned_values() -> None:
 
     study.tell(trial, state=TrialState.PRUNED)
     assert study.trials[-1].value is None
+
+
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_trial_duration_calculation(storage_mode: str) -> None:
+
+    with StorageSupplier(storage_mode) as storage:
+        study = create_study(storage=storage)
+
+        def objective(trial: Trial) -> float:
+            time.sleep(1)
+            x = trial.suggest_int("x", -10, 10)
+            return x
+
+        study.enqueue_trial(params={"x": 1})
+        # Make sure an enqueued trial is created with null datetime_start
+        t_tmp = study.trials[0]
+        assert t_tmp.datetime_start is None
+
+        # Delayed evaluation for enqueued trial
+        time.sleep(2)
+
+        study.optimize(objective, n_trials=2)
+
+        trial = study.ask()
+        values = objective(trial)
+        study.tell(trial, values=values)
+
+        t0 = study.trials[0]
+        assert t0.datetime_start is not None
+        assert t0.datetime_complete is not None
+        duration_error0 = abs((t0.datetime_complete - t0.datetime_start).total_seconds() - 1)
+        assert duration_error0 < 0.1
+
+        t1 = study.trials[1]
+        assert t1.datetime_start is not None
+        assert t1.datetime_complete is not None
+        duration_error1 = abs((t1.datetime_complete - t1.datetime_start).total_seconds() - 1)
+        assert duration_error1 < 0.1
+
+        t2 = study.trials[2]
+        assert t2.datetime_start is not None
+        assert t2.datetime_complete is not None
+        duration_error2 = abs((t2.datetime_complete - t2.datetime_start).total_seconds() - 1)
+        assert duration_error2 < 0.1
```

Comment:
How about using `pytest.approx()` here?
https://docs.pytest.org/en/stable/reference.html#pytest-approx

Reply:
On second thought, I concerned this test case makes the time for test execution longer (seemingly it consumes 5 seconds for each storage backends). We can also reduce the duration of `time.sleep`, but it makes tests fragile.

How about removing this test case? I think the following test case is enough for this purposes.

```python
    with StorageSupplier(storage_mode) as storage:
        study = create_study(storage=storage)

        def objective(trial: Trial) -> float:
            time.sleep(1)
            x = trial.suggest_int("x", -10, 10)
            return x

        study.enqueue_trial(params={"x": 1})
        assert study.trials[0].datetime_start is None

        study.optimize(objective, n_trials=1)
        assert study.trials[0].datetime_start is not None
```

Reply:
Thanks, didn't know `pytest.approx` before. As for the latter one, indeed it takes too long to finish. I agree that we can replace duration calculation with simple `is none` test as we've eliminated repeated `set_trial_state` to `RUNNING`. Will update later.